mlibrary/cozy-sun-bear
# CSB-245: Reflowable scroll mode: unable to access previous section/chapter when text is enlarged

TL;DR: seemed that when you had a CSS override which caused the content to reflow, the scrolling view thrashed with:

- reframe the view based on the initial calculated content dimensions
- adjust the manager viewport scroll to account for views that grew
- tell the rendition that the view was resized
- rendition would then try to `display` the current location
- rendition would then apply the CSS override
- view would be reframed with new content dimensions
- ...and this loops back...
- which would often cause the view that was loaded to scroll **out** of the viewport
- causing it to be unloaded
- but then you'd try scrolling again, which would load it and reframe it to the original dimensions
- repeat

And yes, that's the TL;DR.

- changed the scrolling manager to **not** report views being resized to the rendition
- changed the views so that they never were shrunk when being re-constituted (to avoid the initial/override jitter)

Tested with reflowable and pre-paginated EPUBs.

---

**TO TEST:**

* open any **reflowable EPUB** that can be styled with the Text Size preference
* switch to **Scroll View** 
* open Preferences and **increase the Text Size**
* use the contents to jump into the book
* scroll down through that chapter
* then scroll up fast to reach the previous chapter

You should see: the end of the previous chapter, instead of the container scrolling "back" to the next chapter.
